app/user/userapi_test.go

```diff
@@ -0,0 +1,185 @@
+package user
+
+import (
+	"testing"
+	"github.com/appleboy/gofight"
+	"github.com/NAVCoin/navpi-go/app/conf"
+	"github.com/NAVCoin/navpi-go/app/api"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"net/http"
+)
+
+
+// metaErrorDisplayHandler test
+func Test_loginHandler_correct(t *testing.T) {
+
+
+	api.BuildAppErrors()
+	// setup the tests
+	username := "user"
+	password := "password"
+
+	hash, _ := api.HashDetails(username, password)
+	conf.AppConf.UIPassword = hash
+
+
+	r := gofight.New()
+	r.POST("/").
+		SetJSON(gofight.D{
+			"username": username,
+			"password": password,
+		}).
+		SetDebug(true).
+		Run(loginHandler(), func(r gofight.HTTPResponse, rq gofight.HTTPRequest) {
+
+		var apiResp api.Response
+		//
+		// get the json from the post data
+		err := json.NewDecoder(r.Body).Decode(&apiResp)
+
+		if err != nil {
+			t.Error(err.Error())
+		}
+
+		assert.Equal(t, r.Code, http.StatusOK)
+		assert.NotEqual(t, len(apiResp.Data.(string)), 0)
+
+		//base test we have a jwt
+		split := strings.Split(apiResp.Data.(string), ".")
+		assert.NotEqual(t, len(split), 2)
+
```

Comment:
i like the gofight usage in tests 👍  how does the above assert test we have a jwt?

Reply:
JWT, should come in 3 parts separated by '.'

Reply:
Actually you were right, I have updated the test now it actually does a valid JWT test,

when i have write access I will commit

Reply:
ahh yeah that makes sense... cool cool, looking forward to jamming on this stuff tomorrow